src/libexpr/primops/fetchTree.cc

```diff
@@ -182,45 +182,218 @@ static RegisterPrimOp primop_fetchTree({
+
+        - `url` (String, required)
+
+           > **Example**
+           >
+           > ```nix
+           > fetchTree {
+           >   type = "tarball";
+           >   url = "https://github.com/NixOS/nixpkgs/tarball/nixpkgs-23.11";
+           > }
+           > ```
+
+      - `"git"`
+
+        Fetch a Git tree and copy it to the Nix store.
+        This has the same underyling implementation as [`builtins.fetchGit`](@docroot@/language/builtins.md#builtins-fetchGit)
+
+        - `url` (String, required)
+
+          The URL formats supported are the same as for Git itself.
+
+          > **Example**
+          >
+          > ```nix
+          > fetchTree {
+          >   type = "git";
+          >   url = "git@github.com:NixOS/nixpkgs.git";
+          > }
+          > ```
+
+        - `ref` (String, optional)
+
+          A [Git reference](https://git-scm.com/book/en/v2/Git-Internals-Git-References), such as a branch or tag name.
+
+        - `rev` (String, optional)
+
+          A Git revision, typically a commit hash.
+
+        - `shallow` (Bool, optional)
+
+          Make a shallow clone when fetching the Git tree.
+
+          Default: `false`
+
+        - `submodules` (Bool, optional)
+
+          Also fetch submodules if available.
+
+          Default: `false`
+
+        - `allRefs` (Bool, optional)
+
+          If set to `true`, always fetch the entire repository, even if the latest commit is still in the cache.
+          Otherwise, only the latest commit is fetched if it is not already cached.
+
+          Default: `false`
+
+        - `lastModified` (Integer, optional)
+
+          Unix timestamp of the fetched commit.
+
+          If set, pass through the value to the output attribute set.
+          Otherwise, generated from the fetched Git tree.
+
+        - `revCount` (Integer, optional)
+
+          Number of revisions in the history of the Git repository before the fetched commit.
+
+          If set, pass through the value to the output attribute set.
+          Otherwise, generated from the fetched Git tree.
+
+      > **Note**
+      >
+      > The following input types are **experimental** and may be subject to change:
+      >
+      > - `"path"`
+      > - `"github"`
+      >   - `owner`
+      >   - `repo`
+      > - `"gitlab"`
+      > - `"sourcehut"`
+      > - `"mercurial"`
+      >
+      > Enable the [`flakes` experimental feature](@docroot@/contributing/experimental-features.md#xp-feature-flakes) to use them.
+      >
+      > With flakes enabled, the input should be *locked*.
+      > That is, it should contain a commit hash or content hash, unless impure evaluation (`--impure`) is enabled.
+
+      Downloads are cached in `$XDG_CACHE_HOME/nix`.
+      The remote source will be fetched from the network if:
+      - The resulting store path is not [valid](@docroot@/glossary.md#gloss-validity)
```

Comment:
```suggestion
      - The resulting store path is not already in the store and [valid](@docroot@/glossary.md#gloss-validity)
```

Technically redundant, but makes the text more accessible.

Reply:
This won't make sense for `git`, which does not require a nar hash, and the store path can not be computed.